### DISCUSSION TITLE
icn2153 (not 2053?) supported?

### DISCUSSION THREAD

**Discussion by Westworld** [LAutour/ESP32-HUB75-MatrixPanel-DMA-ICN2053#2] [Q&A]
I have got two boards (128x64) using icn2153 (not icn2053).

Using this library they just display some horizontal lines, then nothing.
(movie: https://github.com/Westworld/HD_DMD_test-/blob/main/matrix.m4v)

should they work - if not - anything I can do to make the library work?

Detailed description (English) about that board is here:
http://kuku.eu.org/?projects/hub75e/index

Source code example to support 2053 (Russian comments):
https://bitbucket.org/nadyrshin_ryu/icn2053_esp32_demo/src/master/

could be modified to support 2153 by replacing one method, as described in the first link above.
Example code (already modified):
https://github.com/Westworld/icn2153_test

This works fine (and prove that my panels work), but runs only with 7fps.
Your library seems to run with 19fps, which would make a big difference for me (panels purchased to be used in a virtual pinball as pinDMD. pinball animations run 20-25 fps, 19fps would be good enough, 7 way too slow.)

Any help to make your library using with 2153 would be great...

**Comment by Westworld** [comment]
Another movie uploaded, showing just one pixel added. I thought this might help. Code used:
  for (int j=0; j<32; j++)
    for (int i=0; i<128; i++) {
    dma_display->drawPixelRGB888(i, j, 255, 0, 0);
    dma_display->flipDMABuffer();
    }

Movie: 
https://github.com/Westworld/HD_DMD_test-/blob/main/pixel.mov

**Reply by LAutour** [reply]
hub75_cfg_t mxconfig = {
    .mx_width = PANEL_WIDTH,
    .mx_height = PANEL_HEIGHT,
...
    .decoder_INT595 = true, // <---------------------- set this row decoder type
  };

**Reply by Westworld** [reply]
super, I did not knew/missed that I need to set this option, major difference. 
Many thanks for this tip!
It draws a line, next line 16 lines above, and so one, for the first 32 lines.
Then it looks as it would draw on the same line again (pixel brightness goes up), then continue on the upper part.

I guess a movie makes more sense.
This is your original demo. For the fullscreen all looks fine, but the plasma looks strange:
https://github.com/Westworld/HD_DMD_test-/blob/main/IMG_3167.m4v

And this draws line by line:
    for (int j=0; j<64; j++) {
    for (int i=0; i<128; i++) {
    dma_display->drawPixelRGB888(i, j, 255, 0, 0);
    }
    dma_display->flipDMABuffer();
    delay(50);
    }

https://github.com/Westworld/HD_DMD_test-/commit/93451e6e23d204621d9969b2f013c90a1d7620e8

I assume I could rearrange y-Position and map that to the correct line, or is there a better way?

**Reply by LAutour** [reply]
post a video for this code:

for (int j=0; j<64; j++)
{
   dma_display->drawFastHLine(10,j,70,127);
   dma_display->flipDMABuffer();
   delay(50); //delay for smooth movement
   dma_display->drawFastHLine(10,j,70,0);
  dma_display->flipDMABuffer();
}

**Reply by Westworld** [reply]
Here is the video:
https://github.com/Westworld/HD_DMD_test-/blob/main/IMG_3171.mov

many thanks for your help!

**Reply by LAutour** [reply]
There are thoughts that it could be, but now my computer has changed, and I need to reconfigure the environment for programming and debugging.

**Comment by Westworld** [comment]
I modified the code to add two lines, such as from 0,0 to 63,63, just to see better where the pixel goes wrong - and (while I don't understand why), now all works fine (https://github.com/Westworld/HD_DMD_test-/blob/main/img_3173.m4v).
Display is working fine with the library, many thanks for your work!
